src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -501,19 +501,27 @@ public IBlockState getExtendedState(@Nonnull IBlockState oldState, @Nonnull IBlo
         }
         
         //check for downflow above corners
-        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, -1)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(0, 0, -1)))
+        boolean n =  isCoveredWithFluid(worldIn, pos.add(-1, 0,  0));
+        boolean s =  isCoveredWithFluid(worldIn, pos.add( 1, 0,  0));
+        boolean w =  isCoveredWithFluid(worldIn, pos.add( 0, 0, -1));
+        boolean e =  isCoveredWithFluid(worldIn, pos.add( 0, 0,  1));
+        boolean nw = isCoveredWithFluid(worldIn, pos.add(-1, 0, -1));
+        boolean ne = isCoveredWithFluid(worldIn, pos.add(-1, 0,  1));
+        boolean sw = isCoveredWithFluid(worldIn, pos.add( 1, 0, -1));
+        boolean se = isCoveredWithFluid(worldIn, pos.add( 1, 0,  1));
+        if(nw || n || w)
         {
             corner[0][0] = 1;
         }
-        if(isCoveredWithFluid(worldIn, pos.add(-1, 0, 0)) || isCoveredWithFluid(worldIn, pos.add(-1, 0, 1)) || isCoveredWithFluid(worldIn, pos.add(0, 0, 1)))
+        if(ne || n || e)
```

Comment:
I know it's already a mix in the existing file, but I think Forge wants "if ("...